mesheryctl/internal/cli/root/system/reset.go

```diff
@@ -48,55 +44,22 @@ func resetMesheryConfig() error {
 	if !userResponse {
 		log.Info("Reset aborted.")
 	} else {
-		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		currCtxName, currCtx, err := utils.GetCurrentContext(tempContext)
 		if err != nil {
-			return errors.Wrap(err, "error processing config")
-		}
-
-		currentContext := mctlCfg.CurrentContext
-
-		// if the user has mentioned a temporary context in the -c flag, change the context and proceed to reset
-		if tempContext != "" {
-			if _, val := mctlCfg.Contexts[tempContext]; val {
-				currentContext = tempContext
-			} else {
-				// if the user specifies a context that is not in the config.yaml file, throw an error and show the available contexts
-				log.Errorf("\n\"%s\" context does not exist. The available contexts are:", tempContext)
-				for context := range mctlCfg.Contexts {
-					log.Errorf("%s", context)
-				}
-				return nil
-			}
+			return errors.Wrap(err, "failed to retrieve current-context")
 		}
 
-		// get the channel and the version of the current context
-		currChannel := mctlCfg.Contexts[currentContext].Channel
-		currVersion := mctlCfg.Contexts[currentContext].Version
-		fileURL := ""
-
 		log.Info("Meshery resetting...\n")
+		log.Printf("Current Context: %s", currCtxName)
+		log.Printf("Channel: %s", currCtx.Channel)
+		log.Printf("Version: %s\n", currCtx.Version)
+		log.Printf("Fetching default docker-compose file at version: %s...\n", currCtx.Version)
 
-		// pull the docker-compose.yaml file corresponding to the current context
-		if currChannel == "edge" {
-			fileURL = "https://raw.githubusercontent.com/layer5io/meshery/master/docker-compose.yaml"
-		} else if currChannel == "stable" {
-			if currVersion == "" {
-				currVersion, err = utils.GetLatestStableReleaseTag()
-				if err != nil {
-					return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to fetch latest stable release tag")))
-				}
-			}
-			fileURL = "https://raw.githubusercontent.com/layer5io/meshery/" + currVersion + "/docker-compose.yaml"
+		err = utils.DownloadDockerComposeFile(tempContext, true)
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch docker-compose file")
```

Comment:
Why do we need to pass in the tempContext? We are already getting the current context in line 47. Once we have the current-context or temp context, we just need to pass that to the `DownloadDockerComposeFile()`function.

Reply:
The current context in line 47 was just for getting the name, channel and version of the context. These values are later logged in the function see line 53-56.